src/EventListener/AddFormatListener.php

```diff
@@ -51,13 +53,15 @@ public function onKernelRequest(GetResponseEvent $event)
             return;
         }
 
-        $this->populateMimeTypes();
-        $this->addRequestFormats($request, $this->formats);
+        $requestAcceptedFormats = $this->getRequestAcceptedFormats($request);
```

Comment:
Actually, this private function can be replaced by `'documentation' === $request->attributes->get('_api_endpoint_type') ? $this->documentationFormats : $this->formats`. It removes 2 function calls in the hot path.